Chinese Academy of Sciences CNIC CSTR.js

```diff
@@ -0,0 +1,194 @@
+{
+	"translatorID": "c070e5a2-4bfd-44bb-9b3c-4be20c50d0d9",
+	"label": "Chinese Academy of Sciences CNIC CSTR",
+	"creator": "cheney",
+	"target": "^https?://www\\.cstr\\.cn",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 97,
+	"inRepository": true,
+	"translatorType": 12,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2022-05-11 08:00:39"
+}
+
+function detectSearch(item) {
+	return !!item.CSTR;
+}
+
+function doSearch(item) {
+	
+	let url;
+	if (item.CSTR) {
+		url = "https://www.cstr.cn/openapi/v2/pid-cstr-service/detail?identifier=" + item.CSTR;
+	}
+	else if (item.query) {
+		url = "https://www.cstr.cn/openapi/v2/pid-common-service/statistics/metadata.search.aggregation?page=1&size=10&query=" + encodeURIComponent(item.query);
+	}
+	Z.debug(url);
+	ZU.doGet(url, parseJSON);
+}
+
+function detectWeb(doc, url) {
+	var searchRe = /^https?:\/\/(?:([^.]+\.))?(?:cstr\.cn)\/(?:search\/cstrDetail\/).*/;

```

Comment:
`.*` at the end isn't needed. I think you could trim this whole test down to

```js
if (searchRe.includes('/search/cstrDetail/') {
```

since the rest of the URL check is handled by the `target` regex in the translator metadata.